table/block_based/block_cache.h

```diff
@@ -156,4 +156,29 @@ template <typename TUse, typename TBlocklike>
 using WithBlocklikeCheck = std::enable_if_t<
     TBlocklike::kCacheEntryRole == CacheEntryRole::kMisc || true, TUse>;
 
+// Helper for the uncache_aggressiveness option
+class UncacheAggressivenessAdvisor {
+ public:
+  UncacheAggressivenessAdvisor(uint32_t uncache_aggressiveness) {
+    assert(uncache_aggressiveness > 0);
+    allowance_ = std::min(uncache_aggressiveness, uint32_t{3});
```

Comment:
Do you mean `std::max`? For example, it feels to me like increasing the setting (above 3) should tolerate more non-useful blocks at the start (when `useful_` is still zero).

Reply:
I meant it this way. I'll add more comments and a unit test for UncacheAggressivenessAdvisor.